### PR TITLE
Use VoodooInput Trackpoint logic for Elan Touchpads

### LINKED ISSUE #2240: Trackpoint on thinkpad doesn't work anymore after "Fixed handleClose not being called by VoodooInput" update
As the title suggests, the trackpoint of a Lenovo Thinkpad (L390) and its buttons no longer work after the last update of VoodooPS2 (https://github.com/acidanthera/VoodooPS2/pull/56). No problem with previous commit.
Elan trackpad works as always.
MacOS 11.7.4; OC 0.9.0
Let me know what I can provide to give you more information.
As always, thanks for your work

### PR DESCRIPTION
**Pull request by averycblack**
Fixes https://github.com/acidanthera/bugtracker/issues/2240
This also simplifies setting properties in the Synaptics service

### CODE REVIEW

File: VoodooPS2Trackpad/VoodooPS2TrackpadCommon.h [hunk 1]
```diff
@@ -11,6 +11,12 @@
 
 #define TEST_BIT(x, y) ((x >> y) & 0x1)
 
+#define DICT_SET_NUM(dict, key, num)                      \
+do {                                                      \
+  OSNumber *val = OSNumber::withNumber(num, 32);          \
+  if (val) { dict->setObject(key, val); val->release(); } \
+} while (0)
+
```
Comment (averycblack) [inline VoodooPS2Trackpad/VoodooPS2TrackpadCommon.h]: Is this better as an inline function here? I'm not sure what the best way to do this is.

Comment (vit9696) [reply VoodooPS2Trackpad/VoodooPS2TrackpadCommon.h]: Well, probably inline function is more C++ way, but I would say either is fine. Performance-wise the inline function should be the same too. If in doubt, I would go the inline function way.

File: VoodooPS2Trackpad/VoodooPS2Elan.cpp [hunk 2]
```diff
@@ -434,6 +412,28 @@ IOReturn ApplePS2Elan::setProperties(OSObject *props) {
     return super::setProperties(props);
 }
 
+void ApplePS2Elan::setTrackpointProperties()
+{
+    // Trackpoint information for VoodooInput
+    OSDictionary *trackpoint = OSDictionary::withCapacity(5);
```
Comment (vit9696) [inline VoodooPS2Trackpad/VoodooPS2Elan.cpp]: Do not we set a lot more than 5 right below? Maybe increase it?

**Review by vit9696 — COMMENTED** [review]
Seems about right to me. Would you merge after CI passes and the comments are addressed?

**Pull request merged**